### PR TITLE
add new shared rubocop config

### PR DESCRIPTION
**Pull request by jreidinger**
changes to previous config will be added as inline comment to this PR

### CODE REVIEW

File: ytools/y2tool/rubocop-1.24.1_yast_style.yml [hunk 1]
```diff
@@ -0,0 +1,213 @@
+AllCops:
+  Exclude:
+    # avoid confusion of rpm spec and gem spec
+    - "**/*.spec"
+  TargetRubyVersion: 2.5 # 2.5 is the SLE15 version, so ensure that we use proper version
+  # By default enable all new cops
+  NewCops: enable
```
Comment (jreidinger) [inline ytools/y2tool/rubocop-1.24.1_yast_style.yml:7]: newly added. Basically default makes updating rubocop more iterative, but it does not fit to way how we update rubocop version.

File: ytools/y2tool/rubocop-1.24.1_yast_style.yml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+# format string is not unified and depending on usage. All three usages has advantages
+# simple style is understandable for translators, but when there are more params it needs names.
+# And template version is needed if there are more params and it needs some formatting.
+Style/FormatStringToken:
+  Enabled: false
+
+Naming/VariableNumber:
+  # do not check numbers usage in symbols as it often come outside of ruby
+  # like `x86_64`
+  CheckSymbols: false
```
Comment (jreidinger) [inline ytools/y2tool/rubocop-1.24.1_yast_style.yml:204]: new item

File: ytools/y2tool/rubocop-1.24.1_yast_style.yml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+# The detection is buggy and result is in potential bugs in Yast::Path handling.
+# It changes
+#    path + "element"
+# to
+#    "#{path}element"
+# so calls path.to_s and append element. Result is wrong path.
+Style/StringConcatenation:
+  Enabled: false
```
Comment (jreidinger) [inline ytools/y2tool/rubocop-1.24.1_yast_style.yml:213]: new item, see explanation

File: ytools/y2tool/rubocop-1.24.1_yast_style.yml [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+# while there is agreement on reducing openstruct usage,
+# it is still used too widely, so disable only per module when
+# it is ready
+Style/OpenStructUse:
```
Comment (jreidinger) [inline ytools/y2tool/rubocop-1.24.1_yast_style.yml:218]: new item

File: ytools/y2tool/rubocop-1.24.1_yast_style.yml [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+  Enabled: false
+
+# Skip warning for constant definition in tests
+# as in rspec we often locate testing data close to respective test
+Lint/ConstantDefinitionInBlock:
```
Comment (jreidinger) [inline ytools/y2tool/rubocop-1.24.1_yast_style.yml:223]: new item

**Review by ancorgs — APPROVED** [review]
LGTM

**Pull request merged**